### PR TITLE
correction for G28 raizes Z axis 3 times

### PR DESCRIPTION
**Pull request by lrpirlet**
In G28, when Z_SAFE_HOMING is disabled and Z_RAISE_BEFORE_HOMING is defined and different from zero, one can see a bad behavior: the Z axis is raised Z_RAISE_BEFORE_HOMING for every axis movement until G28 is done.
This code correct the bad behavior, accessorily, in debug mode, it displays the destination in the `
line_to_destination()` routine.
See #2886 for more information.

### CODE REVIEW

**Comment by Kriechi** [comment]
Thanks for the path - I noticed the same problem!
Although your indentation looks wrong - maybe you could check your spaces/tabs settings?

Please merge!

**Comment by lrpirlet** [comment]
This last pull request includes a better correction suggested by @AnHardt , see #2886

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -2323,6 +2328,7 @@ inline void gcode_G28() {
         #endif
         feedrate = max_feedrate[Z_AXIS] * 60;
         line_to_destination();
+		current_position[Z_AXIS] = destination[Z_AXIS]; //record Z position in the arbitrary system...that will be reset to physical 0,0,0 when all 3 axis have been homed.
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: minor correction: spaces, not tabs.

File: Marlin/Marlin_main.cpp [hunk 2]
```diff
@@ -1163,6 +1163,11 @@ inline void line_to_z(float zPosition) {
   plan_buffer_line(current_position[X_AXIS], current_position[Y_AXIS], zPosition, current_position[E_AXIS], feedrate / 60, active_extruder);
 }
 inline void line_to_destination(float mm_m) {
+  #if ENABLED(DEBUG_LEVELING_FEATURE)
+  if (marlin_debug_flags & DEBUG_LEVELING) {
+ print_xyz("> line_to_destination(), destination", destination);	// displays the destination values
+   }
+  #endif
```
Comment (thinkyhead) [inline Marlin/Marlin_main.cpp]: minor issue: indentation. Replace with:

``` cpp
  #if ENABLED(DEBUG_LEVELING_FEATURE)
    if (marlin_debug_flags & DEBUG_LEVELING) {
      print_xyz("> line_to_destination(), destination", destination);   // displays the destination values
    }
  #endif
```

…or…

``` cpp
  #if ENABLED(DEBUG_LEVELING_FEATURE)
    if (marlin_debug_flags & DEBUG_LEVELING)
      print_xyz("> line_to_destination(), destination", destination);   // displays the destination values
  #endif
```

**Comment by lrpirlet** [comment]
@thinkyhead Thanks, I learned some about indentation... 
 To my understanding the code will perform the same with tabs or space... In fact, I am using notepad++ and I could not spot the differences...
 Is there some formating rules or better a formating program that works under MSwindows (My linux machine is too slow -15 years old- for interactive session)?

**Comment by lrpirlet** [comment]
I did rebase this PR to RCBugFix... I do not want to push, but this PR should help with #1748

**Comment by thinkyhead** [comment]
> I did rebase this PR to RCBugFix

Is that PR still needed since we've integrated recent changes to `RCBugFix` related to Z raising? Did I already merge your updated PR? (Who am I? Why am I here?)

> Is there some formating rules…

I wrote up style guidelines which you can find here:

https://github.com/MarlinFirmware/Marlin/wiki/Coding-Standards/592ec7f019113c33833f8cee1318999022f7b8ed

Ignore the "OUTDATED!" warning. All the guidelines here still apply, and are what I follow for all my code. We will not be using automated formatting, as the updated "Coding-Standards" page claims. That was a very bad idea, a wishful hope that automation would somehow improve things. But this code is meant to be readable and maintainable by humans!

> or better a formating program that works under MSwindows

If you can tell NotePad++ to use 2 spaces instead of a tab character when you press the tab key, simply set that preference. That said, I recommend using a more full-featured extensible text editor like TextMate or Sublime 2 instead. Both are free, but Sublime 2 will nag you occasionally until you buy it. Just hit ESC when it asks you to register.

**Pull request closed without merging**